saltcellar/saltcellar/environment.py

```diff
@@ -0,0 +1,98 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""Gets system environment data."""
+
+import logging
+import os
+import platform
+import subprocess
+import sys
+
+# pylint: disable=invalid-name
+# Get an instance of a logger
+logger = logging.getLogger(__name__)
+
+
+def commit():
+    """Collect the build number for the server.
+
+    :returns: A build number
+    """
+    commit_info = os.environ.get('SALTCELLAR_COMMIT', None)
+    if commit_info is None:
+        try:
+            commit_info = subprocess.check_output(['git',
```

Comment:
Since Python 3.5 the preferred subprocess function is `run()`. You can put a stdout pipe in to capture output. You can also use a check flag if you want an error thrown on failure.

```
result = subprocess.run(["git", "describe", "--always"], stdout=subprocess.PIPE)
return result.stdout.strip().decode('utf-8')
```